langlib/lang.stream/src/main/ballerina/src/lang.stream/stream.bal

```diff
@@ -26,10 +29,29 @@ type PureType anydata|error;
 # + data - data to be published to the stream
 #
 # Each subscriber receives a separate clone of the data.
-public function publish(stream<PureType> strm, PureType data) = external;
+public function publish(stream<PureType> strm, PureType data) {
+    function (PureType)[] funcs = streamManager.getSubscriptionFuncs(strm);
+    int noOfFuncs = arrays:length(funcs);
+    future<any>[] ftrs = [];
+    arrays:forEach(funcs, function(function (PureType) returns () func) {
+        future<any> ftr;
+        if (data is anydata) {
+            ftr = start func(values:clone(data));
+        } else {
+            ftr = start func(data);
+        }
+        ftrs[arrays:length(ftrs)] = ftr;
+    });
+
+    arrays:forEach(ftrs, function(future<any> ftr) {
+        _ = wait ftr;
+    });
```

Comment:
Shall we use the `length`, `forEach` methods on the array itself instead of using imports? 
eg: `ftrs.length();` `ftrs.forEach(fun...)`

Reply:
@irshadnilam it is both the same.. in langlib module, we cannot directly invoke the functions as you mentioned.

Reply:
> in langlib module, we cannot directly invoke the functions as you mentioned

Ah, I didn't know that. :)